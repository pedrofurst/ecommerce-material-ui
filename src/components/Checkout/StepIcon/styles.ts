import { makeStyles, Theme } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) => ({
  root: {
    color: '#eaeaf0',
    display: 'flex',
    height: 22,
    alignItems: 'center',
  },
  active: {
    color: theme.palette.primary.main,
  },
  circle: {
    width: 8,
    height: 8,
    borderRadius: '50%',
    backgroundColor: 'currentColor',
  },
  completed: {
    color: theme.palette.primary.main,
    zIndex: 1,
    fontSize: 18,
  },
}));
