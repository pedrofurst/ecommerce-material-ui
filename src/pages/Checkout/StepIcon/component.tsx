import { StepIconProps } from '@material-ui/core';
import { Check } from '@material-ui/icons';
import clsx from 'clsx';
import React from 'react';
import useStyles from './styles';

function StepIcon(props: StepIconProps) {
  const classes = useStyles();
  const { active, completed } = props;

  return (
    <div
      className={clsx(classes.root, {
        [classes.active]: active,
      })}
    >
      {completed ? (
        <Check className={classes.completed} />
      ) : (
        <div className={classes.circle} />
      )}
    </div>
  );
}

export default StepIcon;
