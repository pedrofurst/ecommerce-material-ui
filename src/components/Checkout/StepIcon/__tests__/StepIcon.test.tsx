import renderer from 'react-test-renderer';
import StepIcon from '../component';

describe('<StepIcon />', () => {
  it('Should render active step icon', () => {
    const component = renderer.create(
      <StepIcon active completed icon={<></>} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
