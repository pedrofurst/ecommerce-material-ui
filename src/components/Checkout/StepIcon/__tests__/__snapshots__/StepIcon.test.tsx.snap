// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepIcon /> Should render active step icon 1`] = `
<div
  className="makeStyles-root-1 makeStyles-active-2"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root makeStyles-completed-4"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</div>
`;
